### PR TITLE
[JENKINS-73280] Migrate to Jakarta EE 9

### PR DESCRIPTION
**Pull request by Dohbedoh**
Started draft for Jakarta EE 9 migration.

### Testing done

TBD

### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -76,8 +76,8 @@
     <dependencies>
       <dependency>
         <groupId>io.jenkins.tools.bom</groupId>
-        <artifactId>bom-2.440.x</artifactId>
-        <version>3307.v2769886db_63b_</version>
+        <artifactId>bom-2.462.x</artifactId>
```
Comment (jglick) [inline pom.xml:79]: note https://github.com/jenkinsci/bom/pull/3677

File: pom.xml [hunk 2]
```diff
@@ -128,6 +128,13 @@
       <artifactId>wordnet-random-name</artifactId>
       <version>1.6</version>
     </dependency>
+    <!--    <dependency>-->
+    <!--      &lt;!&ndash; Waiting for a release of https://github.com/jenkinsci/plugin-pom/pull/1004 &ndash;&gt;-->
+    <!--      <groupId>jakarta.servlet</groupId>-->
+    <!--      <artifactId>jakarta.servlet-api</artifactId>-->
+    <!--      <version>5.0.0</version>-->
+    <!--      <scope>provided</scope>-->
+    <!--    </dependency>-->
```
Comment (jglick) [inline pom.xml:137]:
```suggestion
```

File: src/main/java/com/cloudbees/jenkins/support/SupportAction.java [hunk 3]
```diff
@@ -184,7 +184,7 @@ public boolean isAnonymized() {
     }
 
     @RequirePOST
-    public void doDeleteBundles(StaplerRequest req, StaplerResponse rsp) throws ServletException, IOException {
+    public void doDeleteBundles(StaplerRequest2 req, StaplerResponse2 rsp) throws ServletException, IOException {
```
Comment (jglick) [inline src/main/java/com/cloudbees/jenkins/support/SupportAction.java:187]: BTW better style is to use `HttpResponse` as a return value, in which case you can get rid of the `rsp` parameter and the `ServletException` throw. I am not sure if there is a clean replacement for `req.submittedForm` though, like a `@SomeAnnotation JSONObject form` parameter.

File: src/main/java/com/cloudbees/jenkins/support/SupportMetricsFilter.java
Comment (jglick) [inline src/main/java/com/cloudbees/jenkins/support/SupportMetricsFilter.java:1]: BTW this could be registered as an `HttpServletFilter`.

Comment (Dohbedoh) [reply src/main/java/com/cloudbees/jenkins/support/SupportMetricsFilter.java:1]: But the way we count the activeRequests, we need to wrap the `chain.doFilter` call.

File: pom.xml (hunk 1 above)
Comment (Dohbedoh) [reply pom.xml:79]: Will bring the bom for `2.479.x`.

File: src/main/java/com/cloudbees/jenkins/support/SupportAction.java (hunk 3 above)
Comment (Dohbedoh) [reply src/main/java/com/cloudbees/jenkins/support/SupportAction.java]: Switch to `HttpResponse`

**Comment by Dohbedoh** [comment]
Rebased to fix the merge conflict. Should be good now.

**Pull request merged**